model/src/main/java/com/epam/esm/model/service/impl/GiftCertificateServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.epam.esm.model.service.impl;
+
+import com.epam.esm.model.dao.GiftCertificateDao;
+import com.epam.esm.model.dao.entity.GiftCertificate;
+import com.epam.esm.model.common.SortType;
+import com.epam.esm.model.service.Page;
+import com.epam.esm.model.service.exception.IncorrectArgumentException;
+import com.epam.esm.model.service.exception.NotExistEntityException;
+import com.epam.esm.model.service.GiftCertificateService;
+import com.epam.esm.model.service.converter.impl.GiftCertificateDTOMapper;
+import com.epam.esm.model.service.dto.CertificateDTO;
+import com.epam.esm.model.service.exception.ServiceException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class GiftCertificateServiceImpl implements GiftCertificateService {
+
+    private final GiftCertificateDao certificateDao;
+    private final GiftCertificateDTOMapper dtoMapper;
+
+    public GiftCertificateServiceImpl(GiftCertificateDao certificateDao, GiftCertificateDTOMapper dtoMapper) {
+        this.certificateDao = certificateDao;
+        this.dtoMapper = dtoMapper;
+    }
+
+    @Override
+    @Transactional
```

Comment:
If you want to annotate all methods as @Transcational you can put it at the class level

Reply:
fix